atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -85,6 +89,22 @@ private static ThreadFactory createThreadFactory(String safeLoggablePurpose) {
         return result;
     }
 
+    public List<ListenableFuture<R>> applyBatch(List<T> arguments) {
+        Preconditions.checkState(!closed, "Autobatcher is already shut down");
+        List<ListenableFuture<R>> results = new ArrayList<>();
+        buffer.publishEvents((EventTranslator<DefaultBatchElement<T, R>>[]) arguments.stream().map(argument -> {
+            SettableFuture<R> result = SettableFuture.create();
+            EventTranslator<DefaultBatchElement<T, R>> translator = (refresh, sequence) -> {
+                refresh.result = result;
+                refresh.argument = argument;
+            };
+            results.add(result);
+            return translator;
+        }).collect(Collectors.toList()).toArray());
```

Comment:
This is a bit shady, have you written a test?

Reply:
No tests on anything yet, but can be remedied